### DISCUSSION TITLE
Don´t show "0" if there are NO DB notifications

### DISCUSSION THREAD

**Discussion by vogelor** [filamentphp/filament#11406] [Help] [answered]
### Package

Notifications

### Package Version

v3.2

### How can we help you?

I added DB notifications to my panel - and this work fine. 
My problem is, that if a new DB notification is send, only the badge change from (0) to (1) and thats all. i think my users didn't see this, because the change is to small.
I think it is better to show NO badge if there are NO notifications. then the changing is from "only the bell icon" to "a new number appears". But i can't find out how to do this.
Can please anybody help me?

**Comment by leandrocfe** [comment, marked as answer]
create this view `resources/views/vendor/filament-panels/components/topbar/database-notifications-trigger.blade.php`

```php
<x-filament::icon-button
    :badge="$unreadNotificationsCount ? $unreadNotificationsCount : null"
    color="gray"
    icon="heroicon-o-bell"
    icon-alias="panels::topbar.open-database-notifications-button"
    icon-size="lg"
    :label="__('filament-panels::layout.actions.open_database_notifications.label')"
/>
```

**Reply by vogelor** [reply]
Thank you VERY MUCH.
It works!

**Reply by bilogic** [reply]
@leandrocfe this should be standard behavior out of the box, nobody should be distracted by a brightly colored `0`

**Reply by leandrocfe** [reply]
> @leandrocfe this should be standard behavior out of the box, nobody should be distracted by a brightly colored `0`

send a PR 🤷‍♂️

**Comment by cawecoy** [comment]
Another approach via CSS without touching vendor files:

```
.fi-topbar-database-notifications-btn .fi-icon-btn-badge-ctn {
    display: none;
}
div:has(+ .fi-modal .fi-modal-content) > .fi-topbar-database-notifications-btn .fi-icon-btn-badge-ctn {
    display: block;
}
```
The notification modal has a content container (`.fi-modal-content`) only when there is 1 or more notification.

The downside of this approach is that we are relying on fragile CSS selectors, I guess. If Filament receives updates changing the DOM a little bit, this solution might stop working.

**Reply by stephenlake** [reply]
Just to clarify, the original suggestion is not to edit `~/vendor` files, it's to _create_ a new view in `~/resources/view/vendor/`.
